spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/ExtractDocumentationAsJsonDocletTest.java

```diff
@@ -19,32 +19,89 @@
  */
 package capital.scalable.restdocs.jsondoclet;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-import java.io.File;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.List;
+
+import javax.tools.DocumentationTool.DocumentationTask;
+import javax.tools.JavaFileObject;
+import javax.tools.SimpleJavaFileObject;
+import javax.tools.ToolProvider;
+import static org.junit.Assert.assertTrue;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 
 import org.apache.commons.io.IOUtils;
 import org.json.JSONException;
+import org.junit.Before;
 import org.junit.Test;
 import org.skyscreamer.jsonassert.JSONAssert;
 
 public class ExtractDocumentationAsJsonDocletTest {
 
+    private static final Path SRC_PATH = FileSystems.getDefault().getPath("src/test/resources").toAbsolutePath();
+    private static final Path TGT_PATH = FileSystems.getDefault().getPath("target/test/generated-javadoc-json").toAbsolutePath();
+
     private static final String JSON_PATH =
             "capital/scalable/restdocs/jsondoclet/DocumentedClass.json";
 
-    /**
-     * The test requires that the Doclet is executed before. This is ensured by
-     * the Maven configuration, but not when the test is executed on its own.
-     */
+    private static final List<String> args = List.of(
+        "--release", "9",
+        "-private",
+        "-d", TGT_PATH.toString()
+    );
+
+    @Before
+    public void setup() throws IOException {
+        Files.createDirectories(TGT_PATH);
+    }
+
     @Test
     public void testDocumentedClass() throws IOException, JSONException {
-        String generated = IOUtils.toString(
-                new FileInputStream(new File("target/generated-javadoc-json/" + JSON_PATH)), UTF_8);
-        String expected = IOUtils.toString(
-                this.getClass().getClassLoader().getResourceAsStream(JSON_PATH), UTF_8);
-        JSONAssert.assertEquals(expected, generated, false);
+        Path source = SRC_PATH.resolve("capital/scalable/restdocs/jsondoclet/DocumentedClass.java");
+        Iterable<JavaFileObject> compilationUnits = compilationUnits(source);
+
+        DocumentationTask task = ToolProvider.getSystemDocumentationTool().getTask(null, null, null, ExtractDocumentationAsJsonDoclet.class, args, compilationUnits);
```

Comment:
nice!

Reply:
The spring-auto-restdocs parent POM specifies version 2.11.2 of Jackson Databind. It does not come from Spring. However, I have updated the POM to allow any version after 2.11.1 with the exception of 2.12.0 and added a comment stating that the version specification may be remove once the parent POM specifies a version after 2.12.0.

Reply:
True, the doclet is completely independent of spring 🤦

Reply:
@jmisur What are the next steps to get this merged into your master branch? FWIW, I've been using a custom build that includes this code in my company's build pipeline for almost four months without issue.

Reply:
That's great to hear. I wanted to reply yesterday about possibility to put some example into samples, but let's not overdo it too much. I'll merge this and do a release.